### PR TITLE
add source build for non-x64 linux, and 32-bit binaries for windows

### PR DESCRIPTION
**Pull request by rdeits**

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,52 +1,114 @@
...
+
+force_source_build = lowercase(get(ENV, "CMAKEWRAPPER_JL_BUILD_FROM_SOURCE", "")) in ["1", "true"]
 
-@static if is_linux()
-    download_name = "cmake-$(cmake_version)-Linux-x86_64"
-    filename = "$(download_name).tar.gz"
-    binary = "cmake"
-    binary_path = joinpath("bin", binary)
+process = @static if is_linux()
+    arch = strip(readstring(`arch`))
```
Comment (tkelman) [inline deps/build.jl:71]: are you sure this is always installed, even in minimal environments? why not use the Julia `Sys.ARCH` variable?

Comment (rdeits) [reply deps/build.jl:71]: No reason other than ignorance. https://github.com/rdeits/CMakeWrapper.jl/pull/5

**Comment by tkelman** [comment]
I think these linux binaries might also assume you're using glibc, so they might not work on musl distros like alpine. Maybe a "did the download work" check or error message suggesting what env var to set for the source build fallback would be worth adding?

**Comment by rdeits** [comment]
Good point. See #6

**Pull request merged**